lib/player.js

```diff
@@ -5485,6 +5486,51 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
         shaka.Player.EventName.AbrStatusChanged, data));
   }
 
+  /**
+   * @param {!shaka.util.Error} error
+   * @return {boolean}
+   * @private
+   */
+  canRecoverFromError_(error) {
+    if (error.code !== shaka.util.Error.Code.HTTP_ERROR ||
+      error.category !== shaka.util.Error.Category.NETWORK) {
+      return false;
+    }
+
+    shaka.log.debug('Recoverable NETWORK HTTP_ERROR, trying to recover...');
+
+    // Obtain the active variant and disable it from manifest variants
+    const activeVariantTrack = this.getVariantTracks().find((t) => t.active);
+    goog.asserts.assert(activeVariantTrack, 'Active variant should be found');
+    const manifest = this.manifest_;
+    for (const variant of manifest.variants) {
+      if (variant.id === activeVariantTrack.id) {
+        variant.disabledTime = Date.now() / 1000;
+      }
+    }
+
+    // Apply restrictions in order to disable variants
```

Comment:
If you filter it here, they can never be selected again, so the deactivation time would not be effective.

Reply:
Sorry @avelad I'm newbie with shaka... From where do you think it's the best place to filter? I tested the changes locally and it seems it worked so I thought it was a good "place" ...

Reply:
It might be enough to just set a timer for `maxDisabledTime` seconds, and re-apply restrictions at the end of that timer. And have logic to shut down the timer if the player unloads, etc.
We have a `shaka.util.Timer` utility class for this. Having a single timer at a time and stopping and re-starting it if a new failure happens would probably be fine, since when this method is run it calls `applyRestrictions`.

Reply:
Ok, applied suggestion